### PR TITLE
if/else to test. It was failing with cmake >=3.12

### PR DESCRIPTION
**Pull request by jgsogo**
if/else to test. It was failing with cmake >=3.12

Related docs: https://github.com/conan-io/docs/pull/838

### CODE REVIEW

File: conans/test/utils/build_helpers.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+import subprocess
+from conans.util.files import decode_text
+from semver import SemVer
+
+
+def get_cmake_version():
```
Comment (lasote) [inline conans/test/utils/build_helpers.py]: Move to CMake() as static `get_version()` method. I would prefer to return a `Version` object (from conans model), it is able to compare with strings.

Comment (jgsogo) [reply conans/test/utils/build_helpers.py]: @memsharded, do you agree?

Comment (memsharded) [reply conans/test/utils/build_helpers.py]: Yes, ``CMake.get_version()`` returning a Version makes sense to me

File: conans/client/build/cmake.py [hunk 2]
```diff
@@ -324,3 +325,13 @@ def build(self):
                             dep_str = "${CONAN_%s_ROOT}" % dep.upper()
                             self._conanfile.output.info("Patching paths for %s: %s to %s" % (dep, from_str, dep_str))
                             tools.replace_in_file(path, from_str, dep_str, strict=False)
+
+    @staticmethod
+    def get_version():
+        try:
+            out, err = subprocess.Popen(["cmake", "--version"], stdout=subprocess.PIPE).communicate()
```
[the remark this answers is not in the document]
Comment (jgsogo) [reply conans/client/build/cmake.py]: But maybe I should consider using a `ConanRunner` instance, but it looks like a little bit overkill for this.

**Pull request merged**